src/WebJobs.Script/Scale/FunctionsScaleManager.cs

```diff
@@ -84,11 +105,52 @@ public virtual async Task<ScaleStatusResult> GetScaleStatusAsync(ScaleStatusCont
                 // this can happen if the host is offline
             }
 
-            var vote = GetAggregateScaleVote(votes, context, _logger);
+            List<int> targetScaleVotes = new List<int>();
+
+            if (targetScalersToProcess.Any())
+            {
+                _logger.LogDebug($"{targetScalersToProcess.Count()} target scalers to sample");
+                HostConcurrencySnapshot snapshot = null;
+                try
+                {
+                    snapshot = await _concurrencyStatusRepository.ReadAsync(CancellationToken.None);
+                }
+                catch (Exception exc) when (!exc.IsFatal())
+                {
+                    _logger.LogError(exc, $"Failed to read concurrency status repository");
+                }
+
+                foreach (var targetScaler in targetScalersToProcess)
+                {
+                    try
+                    {
+                        TargetScalerContext targetScaleStatusContext = new TargetScalerContext();
+                        if (snapshot != null)
+                        {
+                            if (snapshot.FunctionSnapshots.TryGetValue(targetScaler.TargetScalerDescriptor.FunctionId, out var functionSnapshot))
+                            {
+                                targetScaleStatusContext.InstanceConcurrency = functionSnapshot.Concurrency;
+                                _logger.LogDebug($"Snapshot dynamic concurrency for target scaler '{targetScaler.TargetScalerDescriptor.FunctionId}' is '{functionSnapshot.Concurrency}'");
+                            }
+                        }
+                        TargetScalerResult result = await targetScaler.GetScaleResultAsync(targetScaleStatusContext);
+                        _logger.LogDebug($"Target worker count for '{targetScaler.TargetScalerDescriptor.FunctionId}' is '{result.TargetWorkerCount}'");
+                        targetScaleVotes.Add(result.TargetWorkerCount);
+                    }
+                    catch (Exception exc) when (!exc.IsFatal())
+                    {
+                        // if a particular target scaler fails, log and continue
+                        _logger.LogError(exc, $"Failed to query scale result for target scaler '{targetScaler.TargetScalerDescriptor.FunctionId}'.");
+                    }
+                }
+            }
 
+            var vote = GetAggregateScaleVote(votes, context, _logger);
```

Comment:
If there are only target scalers and no votes, this will result in a ScaleIn vote, which really isn't consistent if we end up returning a worker count that indicates scale out. One thing we could do is have GetTargetScalersResult return an IEnumerable<TargetScalerVote> where TargetScalerVote contains a ScaleVote and TargetWorkerCount. As these votes are added, ScaleVote can be set correctly based on comparing TargetWorkerCount and WorkerCount. Then, we can pass in the monitorVotes.Union(targetScalerVotes) to GetAggregateScaleVote and it'll aggregate correctly into a single ScaleVote.